include/deal.II/base/symmetric_tensor.h

```diff
@@ -2320,6 +2324,401 @@ Number second_invariant (const SymmetricTensor<2,3,Number> &t)
 
 
 
+/**
+ * Return the eigenvalues of a symmetric tensor of rank 2.
+ *
+ * @relates SymmetricTensor
+ * @author Jean-Paul Pelteret, 2017
+ */
+template <typename Number>
+std::array<Number,1>
+eigenvalues (const SymmetricTensor<2,1,Number> &T);
+
+
+
+/**
+ * Return the eigenvalues of a symmetric tensor of rank 2.
+ * The array of eigenvalues is sorted in descending order.
+ *
+ * For 2x2 tensors, the eigenvalues of tensor $T$ are the roots of
+ * <a href="https://en.wikipedia.org/wiki/Eigenvalue_algorithm#2.C3.972_matrices">the characteristic polynomial</a>
+ * $0 = \lambda^{2} - \lambda*tr(T) + det(T)$
+ * as given by
+ * $\lambda = \frac{tr(T) \pm \sqrt{tr^{2}(T) - 4*det(T)}}{2}$.
+ *
+ * @warning The algorithm employed here determines the eigenvalues by
+ * computing the roots of the characteristic polynomial. In the case that there
+ * exists a common root (the eigenvalues are equal), the computation is
+ * <a href="https://scicomp.stackexchange.com/q/23686">subject to round-off errors</a>
+ * of order $\sqrt{\epsilon}$.
+ * As an alternative, the eigenvectors() function provides a more robust, but costly,
+ * method to compute the eigenvalues of a symmetric tensor.
+ *
+ * @relates SymmetricTensor
+ * @author Jean-Paul Pelteret, 2017
+ */
+template <typename Number>
+std::array<Number,2>
+eigenvalues (const SymmetricTensor<2,2,Number> &T);
+
+
+
+/**
+ * Return the eigenvalues of a symmetric tensor of rank 2.
+ * The array of eigenvalues is sorted in descending order.
+ *
+ * For 3x3 tensors, the eigenvalues of tensor $T$ are the roots of
+ * <a href="https://en.wikipedia.org/wiki/Eigenvalue_algorithm#3.C3.973_matrices">the characteristic polynomial</a>
+ * $0 = \lambda^{3} - \lambda^{2}*tr(T) - \frac{1}{2} \lambda (tr(T^{2}) - tr^{2}(T)) - det(T)$.
+ *
+ * @warning The algorithm employed here determines the eigenvalues by
+ * computing the roots of the characteristic polynomial. In the case that there
+ * exists a common root (the eigenvalues are equal), the computation is
+ * <a href="https://scicomp.stackexchange.com/q/23686">subject to round-off errors</a>
+ * of order $\sqrt{\epsilon}$.
+ * As an alternative, the eigenvectors() function provides a more robust, but costly,
+ * method to compute the eigenvalues of a symmetric tensor.
+ *
+ * @relates SymmetricTensor
+ * @author Jean-Paul Pelteret, 2017
+ */
+template <typename Number>
+std::array<Number,3>
+eigenvalues (const SymmetricTensor<2,3,Number> &T);
+
+
+#ifndef DOXYGEN
+
+namespace internal
+{
+  /**
+   * Tridiagonalize a rank-2 symmetric using the Householder method.
+   * The specialized algorithm implemented here is given in
+   * Kopp, J.
+   * Efficient numerical diagonalization of hermitian 3x3 matrices
+   * International Journal of Modern Physics C, 2008, 19, 523-548
+   * doi: 10.1142/S0129183108012303
+   * arXiv.org preprint: physics/0610206
```

Comment:
doxygen doesn't break lines like you do here. you may want to use punctuation marks to make this look good in the documentation